src/circuit/circuit_frame.rs

```diff
@@ -3055,6 +3175,198 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
 
         let res = AllocatedPtr::from_parts(&res_tag, &val);
 
+        // Next constraints are used for number comparisons
+        ///////////////////////////////////////////////////////////////////////
+        let double_a = constraints::add(&mut cs.namespace(|| "double a"), a, a)?;
+        let double_a_bits = double_a
+            .to_bits_le_strict(&mut cs.namespace(|| "double a lsb"))
+            .unwrap();
+        let lsb_a = double_a_bits.get(0);
+        let bool_val_a = match lsb_a {
+            Some(Boolean::Is(v)) => v.get_value(),
+            _ => {
+                Some(false) // Blank
+            }
+        };
+        let a_is_negative = Boolean::from(
+            AllocatedBit::alloc(cs.namespace(|| "boolean val a"), bool_val_a).unwrap(),
+        );
+
+        let double_b = constraints::add(&mut cs.namespace(|| "double b"), b, b)?;
+        let double_b_bits = double_b
+            .to_bits_le_strict(&mut cs.namespace(|| "double b lsb"))
+            .unwrap();
+        let lsb_b = double_b_bits.get(0);
+        let bool_val_b = match lsb_b {
+            Some(Boolean::Is(v)) => v.get_value(),
+            _ => {
+                Some(false) // Blank
+            }
+        };
+        let b_is_negative = Boolean::from(
+            AllocatedBit::alloc(cs.namespace(|| "boolean val b"), bool_val_b).unwrap(),
+        );
+
+        let both_positive = Boolean::and(
+            &mut cs.namespace(|| "both positive"),
+            &Boolean::not(&a_is_negative),
+            &Boolean::not(&b_is_negative),
+        )?;
+        let both_negative = Boolean::and(
+            &mut cs.namespace(|| "both negative"),
+            &a_is_negative,
+            &b_is_negative,
+        )?;
+        let both_same_sign = constraints::or(
```

Comment:
This uses three constraints total, I think — whereas constructing as `not(xor(…))` would not use any, think.

Reply:
`xor(...)` has the same cost as `or(...)`

Reply:
Yes, my mistake — so 3 vs 1, right?

Reply:
yes, it is done now